### DISCUSSION TITLE
java.io.FileNotFoundException: This file can not be opened as a file descriptor; it is probably compressed

### DISCUSSION THREAD

**Discussion by soruppuAswin** [google/filament#5696] [Ideas]
Hi, I am trying to integrate filament into our application. I have done all the required changes, But, I am currently kind of stuck due to the following error. Can anyone think of what I could be missing?

```
java.io.FileNotFoundException: This file can not be opened as a file descriptor; it is probably compressed
 at android.content.res.AssetManager.nativeOpenAssetFd(Native Method)
 at android.content.res.AssetManager.openFd(AssetManager.java:963)
 at com.walmart.glass.arexperiences.util.MatrixKt.readUncompressedAsset(Matrix.kt:246)
 at com.walmart.glass.arexperiences.renderer.PlaneRenderer.<init>(PlaneRenderer.kt:49)
```

**Comment by romainguy** [comment]
You are trying to mmap a file from assets/. This only works if the file is explicitly marked as uncompressed. This can be done from your Gradle file. For instance to disable compression of gltf files:

```
android {
    ...
    aaptOptions {
       noCompress 'glb'
    }
}
```

**Reply by soruppuAswin** [reply]
It is actually failing where we try to read the filamat files in PlaneRenderer.

`.readUncompressedAsset("materials/textured.filamat")`

This is what I currently have in my gradle file which works as expected in the filament example app.

```
    aaptOptions {
        noCompress 'filamat', 'ktx'
    }
```

I will add glb as you said.

**Reply by romainguy** [reply]
Make sure the files are present in `assets/` then.

**Reply by soruppuAswin** [reply]
They are already in assets

<img width="229" alt="assets" src="https://user-images.githubusercontent.com/46235940/173873528-03763576-0ded-47ec-9de2-11c828e43dc1.png">

**Reply by romainguy** [reply]
The next step would be to inspect the generated APK using Android Studio's APK analyzer to see if the files are there and if they are uncompressed. Alternatively you can modify your reading code to deal with compressed assets (it requires using a regular stream instead of a file descriptor).

**Reply by soruppuAswin** [reply]
I have analyzed the APK and found that the assets(both filamat and glb files) are there in the assets folder. 

Also, When I compare with the filament example app, I am not seeing any difference.

I will also try the alternative approach you suggested of reading the filamat files.

**Reply by soruppuAswin** [reply]
Thank you @romainguy for your help. I am now able to read both filamat and glb files. For filamat, I had to read as a compressed asset to make it work.
